src/Adapters/Cache/InMemoryCache.js

```diff
@@ -0,0 +1,69 @@
+const DEFAULT_CACHE_TTL = 60 * 1000;
+
+
+export class InMemoryCache {
+  constructor({
+    ttl = DEFAULT_CACHE_TTL
+  }) {
+    this.ttl = ttl;
+    this.cache = Object.create(null);
+  }
+
+  get(key) {
+    let record = this.cache[key];
+    if (record == null) {
+      return null;
+    }
+
+    // Has Record and isnt expired
+    if (isNaN(record.expire) || record.expire >= Date.now()) {
+      return record.value;
```

Comment:
The whole point are the promises.

There needs to be two caches in a parse-server one for the applications ``AppCache`` and one for actually cached data  ``CacheController/Adapter`` (_Roles/_Session/etc..).

The ``AppCache`` needs to be synchronous and cant be used remotely (Caches actual JS Objects not JSON), that is where ``InMemoryCache`` comes from, this is so that we don't have two different Cache APIs in parse-server

The promise based architecture is so that users can replace ``CacheAdapter`` with remote caching systems, redis/memcached.

The default ``CacheAdatpter`` (``InMemoryCacheAdapter``) is something that just wraps the synchronous ``InMemoryCache`` with promises.